src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs

```diff
@@ -144,6 +147,41 @@ public bool AllowBindingHeaderValuesToNonStringModelTypes
             set => _allowBindingHeaderValuesToNonStringModelTypes.Value = value;
         }
 
+        /// <summary>
+        /// Gets or sets a value that determines if model bound action parameters, controller properties, page handler
+        /// parameters, or page model properties are validated (in addition to validating their elements or
+        /// properties). If set to <see langword="true"/>, <see cref="BindRequiredAttribute"/> and
+        /// <c>ValidationAttribute</c>s on these top-level nodes are checked. Otherwise, such attributes are ignored.
+        /// </summary>
+        /// <value>
+        /// The default value is <see langword="true"/> if the version is
+        /// <see cref="CompatibilityVersion.Version_2_1"/> or later; <see langword="false"/> otherwise.
+        /// </value>
+        /// <remarks>
+        /// <para>
+        /// This property is associated with a compatibility switch and can provide a different behavior depending on
+        /// the configured compatibility version for the application. See <see cref="CompatibilityVersion"/> for
+        /// guidance and examples of setting the application's compatibility version.
+        /// </para>
+        /// <para>
+        /// Configuring the desired value of the compatibility switch by calling this property's setter will take
+        /// precedence over the value implied by the application's <see cref="CompatibilityVersion"/>.
+        /// </para>
+        /// <para>
+        /// If the application's compatibility version is set to <see cref="CompatibilityVersion.Version_2_0"/> then
+        /// this setting will have the value <see langword="false"/> unless explicitly configured.
+        /// </para>
+        /// <para>
+        /// If the application's compatibility version is set to <see cref="CompatibilityVersion.Version_2_1"/> or
+        /// higher then this setting will have the value <see langword="true"/> unless explicitly configured.
+        /// </para>
+        /// </remarks>
+        public bool AllowValidatingTopLevelNodes
```

Comment:
Should have tests covering all uses of this new option with it both `true` and `false`. Let me know if I missed something.

Reply:
Since this is a compat switch, consider whether or not the functional tests set it. One looming issue that I realize I missed, is that I had intended to force all of our functional tests to set the compat mode to 2.1. I realized earlier this week that I forgot to do this.

The rationale for this is that we thing the new version is the best version. We care the most about the best version being the best it it.

Reply:
One functional test has (oddly) started to fail. Now seeing that locally as well.